projects/concurrency/cache-computing/cache.go

```diff
@@ -0,0 +1,65 @@
+package cachecomputing
+
+import (
+	"concurrency/cache"
+	"sync"
+)
+
+type Creator[K comparable, V any] func(K) V
+
+type ComputingCache[K comparable, V any] struct {
+	cache                   cache.Cache[K, V]
+	creator                 Creator[K, V]
+	pendingKeys             map[K]chan V
+	numberOfWaitingRoutines map[K]int
+	mu                      sync.Mutex
+}
+
+func NewComputingCache[K comparable, V any](entryLimit int, creator Creator[K, V]) *ComputingCache[K, V] {
+	return &ComputingCache[K, V]{
+		cache:                   cache.NewCache[K, V](entryLimit),
+		creator:                 creator,
+		numberOfWaitingRoutines: map[K]int{},
+		pendingKeys:             map[K]chan V{},
+	}
+}
+
+func (c *ComputingCache[K, V]) Get(key K) V {
+	value, present := c.cache.Get(key)
+	if present {
+		return *value
+	}
+
+	c.mu.Lock()
+	if channel, ok := c.pendingKeys[key]; ok {
```

Comment:
There's a small race here, that some other goroutine may have transitioned our key from "pending" to "value known" between the `Get` check above, and the `pendingKeys` check here, which would cause this goroutine to re-calculate the value even though it's already known - how do you think we can handle that?